### ISSUE TITLE
k8s-triage-robot should not be closing important bugs

### ISSUE THREAD

**Issue by wojtek-t** [kubernetes/test-infra#25967]
We were just chatting with @liggitt about reliability bar and one of the AIs from the discussions was to ensure that important bugs actually aren't closed.

Basically, any issue that is marked with:
kind/bug
priority/{important-soon,important-longterm,critical-urgent}

should not be touched by k8s-triage-robot for marking as state/rotten/closed

Couple different jobs in this file have to be updated to do this.
https://github.com/kubernetes/test-infra/blob/705997b53f349731aa03c355c50637af574a2917/config/jobs/kubernetes/sig-k8s-infra/trusted/sig-contribex-k8s-triage-robot.yaml#L133

@kubernetes/sig-contributor-experience 
/help wanted

**Comment by liggitt** [comment]
xref https://github.com/kubernetes/kubernetes/issues/103151

I agree with this scoped starting point. A confirmed bug that is marked important/critical should not be autoclosed.

**Comment by liggitt** [comment]
I could actually see the stale/rotten labels being useful/interesting to indicate inactivity/neglect, but not auto-closing

**Comment by wojtek-t** [comment]
> I could actually see the stale/rotten labels being useful/interesting to indicate inactivity/neglect, but not auto-closing

+1 - didn't think about it but it makes perfect sense


Also thanks for cross-referencing.
Adding some folks here explicitly then:
@dims @BenTheElder @sftim @ehashman @spiffxp

**Comment by liggitt** [comment]
for reference, here's a query of the closed `lifecycle/rotten` bugs with important or critical priority: https://github.com/kubernetes/kubernetes/issues?q=is%3Aissue+is%3Aclosed+label%3Alifecycle%2Frotten+label%3Akind%2Fbug+label%3Apriority%2Fimportant-longterm%2Cpriority%2Fimportant-soon%2Cpriority%2Fcritical-urgent+

**Comment by wojtek-t** [comment]
Should we just go ahead and reopen them?
[Not now, but after changing the bot]

**Comment by BenTheElder** [comment]
The bot does search queries so changing them is trivial, but up to contributor experience to approve.

I think the problem is "confirmed" bug.

E.g. you could just apply `/lifecycle frozen` to accepted bugs but anyone can do this. It's also true for `/kind bug` and `/priority critival-urgent` though.

**Comment by sftim** [comment]
I'd be happy to see accepted issues never closing, or rotting with a _much_ longer interval (eg 12 months). We can still explicitly freeze key issues.

**Comment by mrbobbytables** [comment]
I agree that keeping their lifecycle state is useful knowledge, and I believe bugs labeled with `triage/accepted` not closing is an acceptable middle-ground. Only org members can use the command, so I think there is sufficient gating to prevent abuse. 

IMO - It's probably worth sending to the contribex/k-dev mailing lists+community meeting this week for more broader discussion.

**Comment by jberkus** [comment]
.... or even adding it to the Community Meeting agenda.

**Comment by wojtek-t** [comment]
We just discussed that during the community meeting.

The outcome was that:
1. we want to proceed with not closing issues that are what's originally proposed + triage/accepted
2. we want to ensure that SIGs are actually triaging issues 
3. we want to ensure that issues that don't have SIG assigned are also triaged
4. we want to additionally provide some customization for timelines (e.g. enhancements repo wants more than 3 months for getting stale)

All of those are valid requests, but we shouldn't block (1) on the other.


So

here is the query (for k/k example) of the issues that should NOT be closed:

https://github.com/kubernetes/kubernetes/issues?q=is%3Aissue+-label%3Alifecycle%2Ffrozen+label%3Alifecycle%2Frotten+label%3Akind%2Fbug+label%3Atriage%2Faccepted+label%3Apriority%2Fcritical-urgent%2Cpriority%2Fimportant-soon%2Cpriority%2Fimportant-longterm+

here is the query that is used currently:
https://github.com/kubernetes/kubernetes/issues?q=is%3Aissue+-label%3Alifecycle%2Ffrozen+label%3Alifecycle%2Frotten


But I didn't yet figure out how to get the diff from those two. Any hints?

**Comment by cblecker** [comment]
The configuration of the bot is here: https://github.com/kubernetes/test-infra/blob/master/config/jobs/kubernetes/sig-k8s-infra/trusted/sig-contribex-k8s-triage-robot.yaml

**Comment by cblecker** [comment]
The process of recording consensus and notifying folks is here: https://github.com/kubernetes/community/blob/master/sig-contributor-experience/charter.md#cross-cutting-and-externally-facing-processes

**Comment by wojtek-t** [comment]
@cblecker - thanks; the announcement have been sent in:
https://groups.google.com/a/kubernetes.io/g/leads/c/PYjDxRh8ghQ

**Comment by wojtek-t** [comment]
/remove-lifecycle stale

There are still things to do described in:
https://github.com/kubernetes/test-infra/issues/25967#issuecomment-1105612832

**Comment by jberkus** [comment]
Can you render that as a checklist in the original issue comment so that everyone knows what still needs to be done?  Thanks!

**Comment by wojtek-t** [comment]
> Can you render that as a checklist in the original issue comment so that everyone knows what still needs to be done? Thanks!

Done

**Comment by marcindulak** [comment]
:eyes:

**Comment by BenTheElder** [comment]
@wojtek-t do we have any further changes to discuss here?

we have a few related issues and I think it's gotten scattered over time ...
